src/Parallel/Callback.hpp

```diff
@@ -0,0 +1,106 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines Parallel::Callback.
+
+#pragma once
+
+#include <pup.h>
+#include <tuple>
+#include <utility>
+
+#include "Parallel/Invoke.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Parallel/CharmPupable.hpp"
+
+namespace Parallel {
+/// A class that holds a function that can be invoked at a later time.
+/// Used inside MutableGlobalCache.
```

Comment:
I don't think we need to document what uses a class unless this is the only place it should be used.  And if it should only be used inside MutableGlobalCache then its visibility should be limited either by making this a subclass of MutableGlobalCache or have its members be private and exposed through friendship.  But I don't immediately see a reason why the use of this should be restricted to within MutableGlobalCache